explainaboard/analysis/feature.py

```diff
@@ -1,123 +1,265 @@
-        self.dtype = "dict"
+    @property
+    def feature(self) -> dict[str, FeatureType]:
+        return self._feature
 
+    def serialize(self) -> dict[str, SerializableData]:
+        """See Serializable.serialize."""
+        data = super().serialize()
+        data["feature"] = self._feature
+        return data
+
+    @classmethod
+    def deserialize(cls, data: dict[str, SerializableData]) -> Serializable:
+        """See Serializable.deserialize."""
+        feature = {
+            # See https://github.com/python/mypy/issues/4717
+            k: narrow(FeatureType, v)  # type: ignore
+            for k, v in narrow(dict, data["feature"]).items()
+        }
+
+        return cls(
+            feature,
+            description=_get_value(str, data, "description"),
+            func=None,
+            require_training_set=_get_value(bool, data, "require_training_set"),
+        )
 
-@dataclass
-class Position(FeatureType):
-    positions: Optional[list] = None
 
-    def __post_init__(self):
-        super().__post_init__()
-        self.cls_name: str = "Position"
+# @dataclass
+# class Position(FeatureType):
+#    positions: Optional[list] = None
+#
+#    def __post_init__(self):
+#        super().__post_init__()
+#        self.cls_name: str = "Position"
 
 
-@dataclass
+@final
+@_feature_type_registry.register("Value")
 class Value(FeatureType):
+    def __init__(
+        self,
+        max_value: int | float | None = None,
+        min_value: int | float | None = None,
+        dtype: str | None = None,
+        description: str | None = None,
+        func: Callable[..., Any] | None = None,
+        require_training_set: bool | None = None,
+    ) -> None:
+        """Initializes Value object.
 
-    # the maximum value (inclusive) of a feature with the
-    # dtype of `float` or `int`
-    max_value: Optional[float | int] = None
-    # the minimum value (inclusive) of a feature with the
-    # dtype of `float` or `int`
-    min_value: Optional[float | int] = None
-
-    def __post_init__(self):
-        super().__post_init__()
-        if self.dtype == "double":  # fix inferred type
-            self.dtype = "float64"
-        if self.dtype == "float":  # fix inferred type
-            self.dtype = "float32"
-
-
-FEATURETYPE_REGISTRY = {
-    "FeatureType": FeatureType,
-    "Sequence": Sequence,
-    "Dict": Dict,
-    "Position": Position,
-    "Value": Value,
-}
+        Args:
+            max_value: The maximum value (inclusive) of values with int/float dtype.
+            min_value: The minimum value (inclusive) of values with int/float dtype.
+            dtype: See FeatureType.__init__.
+            description: See FeatureType.__init__.
+            func: See FeatureType.__init__.
+            require_training_set: See FeatureType.__init__.
+        """
+        # Fix inferred types.
+        if dtype == "double":
+            dtype = "float64"
+        elif dtype == "float":
+            dtype = "float32"
```

Comment:
@neubig Are these conversion really needed? This confuses users by increasing ambiguity of `dtype` values.

Reply:
I think we can remove it.

Reply:
@pfliu-nlp Thanks. I have seen some callers give double/float to Value, which should be fixed too in this PR.

Reply:
Yeah, I think this is OK to remove.

Reply:
Other question: what is the valid values of this field? Since there is no any restriction currently, we can set any "dtypes" to Value, which looks not a desired behavior.

Reply:
Since removing this block involves many potential changes, I will create other pull request.

Reply:
Just checked the code: it seems this is the only use case for the current version of main branch: https://github.com/neulab/ExplainaBoard/blob/8c514c3d81a079d967d208f8bc330c2f202620bb/explainaboard/processors/processor.py#L64

Reply:
Created an issue: #466